libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -110,228 +90,196 @@ export class OrchestratorAdaptiveRecognizer extends Recognizer implements Orches
+                throw new Error(`Model load failed.`);
             }
-
-            const instance = {
-                startIndex: entityResult['start'],
-                endIndex: entityResult['end'],
-                score: 1.0,
-                text: entityResult['text'],
-                type: entityResult['type'],
-                resolution: entityResult['resolution'],
-            };
-            instanceData.push(instance);
+            OrchestratorAdaptiveRecognizer.orchestrator = orchestrator;
+            console.timeEnd('Model load');
         }
-    }
 
-    /**
-     * @private
-     */
-    private Initialize() {
-        if (OrchestratorAdaptiveRecognizer.orchestrator === null || this.resolver == null) {
-            if (this._modelPath == null) {
-                throw new Error(`Missing "ModelPath" information.`);
-            }
-
-            if (this._snapshotPath == null) {
-                throw new Error(`Missing "ShapshotPath" information.`);
-            }
-            const fullModelPath = resolve(this._modelPath);
+        if (!this._resolver) {
             const fullSnapshotPath = resolve(this._snapshotPath);
-            if (!existsSync(fullModelPath)) {
-                throw new Error(`Model folder does not exist at ${fullModelPath}.`);
-            }
             if (!existsSync(fullSnapshotPath)) {
                 throw new Error(`Snapshot file does not exist at ${fullSnapshotPath}.`);
             }
+            // Load the snapshot
+            const snapshot: Uint8Array = readFileSync(fullSnapshotPath) as Buffer;
 
-            if (OrchestratorAdaptiveRecognizer.orchestrator == null) {
-                console.time('Model load');
-                // Create orchestrator core
-                OrchestratorAdaptiveRecognizer.orchestrator = new oc.Orchestrator();
-                if (OrchestratorAdaptiveRecognizer.orchestrator.load(fullModelPath) === false) {
-                    throw new Error(`Model load failed.`);
-                }
-                console.timeEnd('Model load');
-            }
-
-            if (this.resolver == null) {
-                // Load the snapshot
-                const encoder: any = new TextEncoder();
-                const fileContent: string = ReadText.readSync(fullSnapshotPath);
-                const snapshot: Uint8Array = encoder.encode(fileContent);
-
-                // Load snapshot and create resolver
-                this.resolver = OrchestratorAdaptiveRecognizer.orchestrator.createLabelResolver(snapshot);
-            }
+            // Load snapshot and create resolver
+            this._resolver = OrchestratorAdaptiveRecognizer.orchestrator.createLabelResolver(snapshot);
         }
     }
 }
+
+/*
+ * This function generates a GUID-like random number that should be sufficient for our purposes of tracking
+ * instances of a given waterfall dialog.
+ * Source: https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
+ */
+function generate_guid() {
```

Comment:
Let's get this function added to `botbuilder-stdlib` I see it all over the place.

Reply:
Yes, shall we send out another PR for this?

Reply:
Sure, sounds good.

Reply:
Added https://github.com/microsoft/botbuilder-js/issues/3160 to track.